### ISSUE TITLE
amazon.aws.ec2_vpc_nat_gateway tags and purge_tags are unsupported parameters

### ISSUE THREAD

**Issue by KSpakes87** [ansible-collections/amazon.aws#1612]
### Summary

When using the amazon.aws.ec2_vpc_nat_gateway module I cannot use tags and purge_tags parameters. I get the following message.

fatal: [local]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (ec2_vpc_nat_gateway) module: purge_tags, tags Supported parameters include:
allocation_id, aws_access_key, aws_ca_bundle, aws_config, aws_secret_key, client_token, debug_botocore_endpoint_logs, ec2_url, eip_address, if_exist_do_not_cr
eate, nat_gateway_id, profile, region, release_eip, security_token, state, subnet_id, validate_certs, wait, wait_timeout"}

### Issue Type

Bug Report

### Component Name

amazon.aws.ec2_vpc_nat_gateway

### Ansible Version

```console (paste below)
$ ansible --version

```
ansible 2.10.8
  config file = /home/ubuntu/playbooks/acmecorp/ansible.cfg
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]

### Collection Versions

```console (paste below)
$ ansible-galaxy collection list
```
amazon.aws                1.4.0
ansible.netcommon         1.5.0
ansible.posix             1.1.1
ansible.windows           1.4.0
arista.eos                1.3.0
awx.awx                   14.1.0
azure.azcollection        1.4.0
check_point.mgmt          1.0.6
chocolatey.chocolatey     1.0.2
cisco.aci                 1.1.1
cisco.asa                 1.0.4
cisco.intersight          1.0.10
cisco.ios                 1.3.0
cisco.iosxr               1.2.1
cisco.meraki              2.2.0
cisco.mso                 1.1.0
cisco.nso                 1.0.3
cisco.nxos                1.4.0
cisco.ucs                 1.6.0
cloudscale_ch.cloud       1.3.1
community.aws             1.3.0
community.azure           1.0.0
community.crypto          1.4.0
community.digitalocean    1.0.0
community.docker          1.2.2
community.fortios         1.0.0
community.general         1.3.6
community.google          1.0.0
community.grafana         1.1.0
community.hashi_vault     1.1.0
community.hrobot          1.1.0
community.kubernetes      1.1.1
community.kubevirt        1.0.0
community.libvirt         1.0.0
community.mongodb         1.2.0
community.mysql           1.2.0
community.network         1.3.2
community.okd             1.0.0
community.postgresql      1.1.1
community.proxysql        1.0.0
community.rabbitmq        1.0.1
community.routeros        1.1.0
community.skydive         1.0.0
community.vmware          1.7.0
community.windows         1.3.0
community.zabbix          1.2.0
containers.podman         1.4.1
cyberark.conjur           1.1.0
cyberark.pas              1.0.5
dellemc.os10              1.0.2
dellemc.os6               1.0.6
dellemc.os9               1.0.3
f5networks.f5_modules     1.7.1
fortinet.fortimanager     1.0.5
fortinet.fortios          1.1.8
frr.frr                   1.0.3
gluster.gluster           1.0.1
google.cloud              1.0.2
hetzner.hcloud            1.2.1
ibm.qradar                1.0.3
infinidat.infinibox       1.2.4
junipernetworks.junos     1.3.0
mellanox.onyx             1.0.0
netapp.aws                20.9.0
netapp.elementsw          20.11.0
netapp.ontap              20.12.0
netapp_eseries.santricity 1.1.0
netbox.netbox             1.2.1
ngine_io.cloudstack       1.2.0
ngine_io.exoscale         1.0.0
ngine_io.vultr            1.1.0
openstack.cloud           1.2.1
openvswitch.openvswitch   1.1.0
ovirt.ovirt               1.3.0
purestorage.flasharray    1.6.2
purestorage.flashblade    1.4.0
servicenow.servicenow     1.0.4
splunk.es                 1.0.2
theforeman.foreman        1.5.1
vyos.vyos                 1.1.1
wti.remote                1.0.1

### AWS SDK versions

```console (paste below)
$ pip show boto boto3 botocore
```
Name: boto
Version: 2.49.0
Summary: Amazon Web Services Library
Home-page: https://github.com/boto/boto/
Author: Mitch Garnaat
Author-email: mitch@garnaat.com
License: MIT
Location: /usr/local/lib/python3.10/dist-packages
Requires:
Required-by:
---
Name: boto3
Version: 1.26.136
Summary: The AWS SDK for Python
Home-page: https://github.com/boto/boto3
Author: Amazon Web Services
Author-email:
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: botocore, jmespath, s3transfer
Required-by:
---
Name: botocore
Version: 1.29.136
Summary: Low-level, data-driven core of boto 3.
Home-page: https://github.com/boto/botocore
Author: Amazon Web Services
Author-email:
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: jmespath, python-dateutil, urllib3
Required-by: boto3, s3transfer

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```
DEFAULT_PRIVATE_KEY_FILE(/home/ubuntu/playbooks/acmecorp/ansible.cfg) = /home/ubuntu/.ssh/acmecorp.pem

### OS / Environment

Ubuntu Server 22.04

### Steps to Reproduce

```yaml (paste below)

```
---
- name: Create NGW with New Elastic IP
  ec2_vpc_nat_gateway:
    if_exist_do_not_create: True
    region: "{{ region }}"
    state: present
    subnet_id: "{{ subnet_id }}"
    purge_tags: False
    tags:
      Name: "{{ ngw_name }}"
    wait: "{{ wait }}"
  register: ngw

### Expected Results

Create NGW with an Elastic IP with a tag of "Name: NGW_Name"

### Actual Results

```console (paste below)

```
fatal: [local]: FAILED! => {"changed": false, "msg": "Unsupported parameters for (ec2_vpc_nat_gateway) module: purge_tags, tags Supported parameters include:
allocation_id, aws_access_key, aws_ca_bundle, aws_config, aws_secret_key, client_token, debug_botocore_endpoint_logs, ec2_url, eip_address, if_exist_do_not_cr
eate, nat_gateway_id, profile, region, release_eip, security_token, state, subnet_id, validate_certs, wait, wait_timeout"}

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by KSpakes87** [comment]
Just to note, there is a double space below `tags` on `Name: "{{ ngw_name }}". Doesn't look like the post put that in there.

**Comment by alinabuzachis** [comment]
@KSpakes87 You seem to be using an old version of the amazon.aws and community.aws collections. The ec2_vpc_nat_gateway module has already been migrated to the amazon.aws collection in version 2.0. We recently released amazon.aws 6.1.0. Can you please update the amazon.aws collection version and let me know? Thank you.

**Comment by KSpakes87** [comment]
@alinabuzachis Looks like that was the issue. This was a fresh install of Ansible, so it might not be coming with the updated version of `amazon.aws`. 

I am now getting a warning when I run `ansible-playbook` now. However, this is not the place for that and may just be needing another update.

**Comment by alinabuzachis** [comment]
@KSpakes87 Glad you have solved. I'm going to close this issue right now, but feel free to open a new one if something else arises. That you!

**Comment by tremble** [comment]
> This was a fresh install of Ansible, so it might not be coming with the updated version of amazon.aws.

Beware, your responses list you as using Ansible 2.10.  This version of Ansible-core is no longer supported by either upstream (ansible-core 2.13 is the oldest supported release), or by this collection (ansible-2.12 is the oldest this collection will work with).

Ansible 8.0.0 (based on ansible-core 2.15) was recently released and should come with release 6.0.1 of this collection (latest release is 6.1.0, with only some minor changes)
